cognite/neat/graph/transformers/_value_type.py

```diff
@@ -0,0 +1,110 @@
+from typing import cast
+
+from rdflib import XSD, Graph, URIRef
+
+from cognite.neat.issues._base import IssueList
+from cognite.neat.rules.analysis import InformationAnalysis
+from cognite.neat.rules.models._rdfpath import SingleProperty
+from cognite.neat.rules.models.data_types import AnyURI, DataType
+from cognite.neat.rules.models.entities import (
+    ClassEntity,
+    MultiValueTypeInfo,
+)
+from cognite.neat.rules.models.information import InformationRules
+
+from ._base import BaseTransformer
+
+
+class SplitMultiValueProperty(BaseTransformer):
+    description: str = (
+        "SplitMultiValueProperty is a transformer that splits a "
+        "multi-value property into multiple single-value properties."
+    )
+    _use_only_once: bool = True
+    _need_changes = frozenset({})
+
+    _object_property_template: str = """SELECT ?s ?o WHERE{{
+
+                                ?s a <{subject_uri}> .
+                                ?s <{property_uri}> ?o .
+                                ?o a <{object_uri}> .
+
+                            }}"""
+
+    _datatype_property_template: str = """SELECT ?s ?o WHERE {{
+
+                                ?s a <{subject_uri}> .
+                                ?s <{property_uri}> ?o .
+                                FILTER (datatype(?o) = <{datatype}>)
+
+                                }}"""
+
+    def transform(self, graph: Graph) -> None:
+        rules = self._infer_rules(graph)
```

Comment:
Isn't this an expensive operation to do? Isn't there simpler SPARQL queries to find multivalued properties?

I am afraid that for larger graph this will be very time consuming.

Reply:
outdated :)